app/src/main/java/com/example/pokemon/ui/MainActivity.kt

```diff
@@ -57,16 +67,22 @@ class MainActivity : AppCompatActivity(), MainContract.MainView,
 
         //open favorites activity
         favorites.setOnClickListener {
-            val intent = Intent(this@MainActivity, FavoritesActivity::class.java)
+            val intent = Intent(this, FavoritesActivity::class.java)
             startActivity(intent)
 
         }
 
     }
 
+    fun isConnectedToNetwork() = Common.isConnectedToNetwork(applicationContext)
+
     override fun onResume() {
         super.onResume()
-        mainPresenterImpl?.getPokemonDetailsFromDb()
+
+        if(!isConnectedToNetwork() || firstTimeOpened)
+            presenter.getPokemonDetailsFromDb()
+
+        firstTimeOpened = true
```

Comment:
set this flag in onCreate not onResume and check if ur code works or not

Reply:
No, it doesn't work.